src/main/java/Duke.java

```diff
@@ -1,10 +1,135 @@
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
-    }
-}
+import java.util.Scanner;
+public class Duke {
+    public static Scanner input = new Scanner(System.in);
+    public static Task[] list= new Task[100];
+    public static String[] words = new String[10];
+    public static String[] phrases = new String[10];
+    public static int currentTaskNum = 0;
+    public static String lineBreaker = "____________________________________________________________\n";
+    public static String errorMessage = lineBreaker + "Invalid input. Please try again!\n" + lineBreaker;
+    public static String logo = " ____        _        \n"
+            + "|  _ \\ _   _| | _____ \n"
+            + "| | | | | | | |/ / _ \\\n"
+            + "| |_| | |_| |   <  __/\n"
+            + "|____/ \\__,_|_|\\_\\___|\n";
+    public static String greeting = lineBreaker +
+            " Hi! I'm Duke\n" +
+            " What can I do for you?\n" +
+            lineBreaker;
+    public static String farewellMessage =
+            " Bye. Hope to see you again soon!\n" + lineBreaker;
+
+    public boolean shouldExit = false;
+
+    public void setShouldExit () {
+        this.shouldExit = true;
+    }
+
+    public static void printList(int currentTaskNum){
+        int currentPrintedTask = 0;
+        int tempPos = currentTaskNum;
+        if (tempPos == 0) {
+            System.out.println(lineBreaker);
+            System.out.println("No Task!");
+        } else {
+            System.out.println(lineBreaker);
+            while (tempPos > 0) {
+                System.out.println(currentPrintedTask + 1 + ". " + list[currentPrintedTask].toString());
+                currentPrintedTask++;
+                tempPos--;
+            }
+        }
+        System.out.println(lineBreaker);
+    }
+
+    public void handleError() {
+        System.out.println(errorMessage);
+    }
+
+    public void printTotalTasks(int currentTaskNum) {
+        if (currentTaskNum == 1) {
+            System.out.println("Now you have " + currentTaskNum + " task in the list.");
+        } else {
+            System.out.println("Now you have " + currentTaskNum + " tasks in the list.");
+        }
+    }
+
+    public int getMarkPosition(String userInput) {
+        words = userInput.split(" ");
+        return Integer.parseInt(words[1]) - 1;
+    }
+    
+    public void toggleMark(int posOfMark, boolean shouldMarkAsDone) {
+        if (!(posOfMark >= 0 && posOfMark <= currentTaskNum)) {
+            handleError();
+        } else {
+            if (shouldMarkAsDone) {
+                list[posOfMark].markAsDone();
+            } else {
+                list[posOfMark].markAsUndone();
+            }
+        }
+    }

```

Comment:
Perhaps you could add documentation using JavaDoc so readers can understand what this function does.